### PR TITLE
[Processing] Vector Table chapter

### PR DESCRIPTION
**Pull request by ghtmtt**
Final PR for the related Processing Vector Table chapter

### CODE REVIEW

**Comment by ghtmtt** [comment]
I have no idea why travis is complaining about the `python` syntax code-block. Local compilation works fine and the directive is correct (http://www.sphinx-doc.org/en/stable/markup/code.html#line-numbers).

Should I remove it?

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst [hunk 1]
```diff
@@ -404,246 +146,150 @@ Parameters
...
 
 Parameters
 ..........
 
 ``Input layer`` [vector: any]
-  Source layer.
+  Input source layer
 
 ``Fields mapping`` [fieldsmapping]
   Output fields definitions
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: Any chance for more litterature, ie more information on the fields of this widget (which are actually the real parameters of this algorithm)

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst [hunk 2]
```diff
@@ -404,246 +146,150 @@ Parameters
...
 
 Text to float
 -------------
-
-Description
-...........
-
-<put algorithm description here>
+Modifies the type of a given attribute in a vector layer, converting a text attribute
+containing numeric strings into a numeric attribute.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: what happens if it's not all about numeric strings in the original field?

Comment (DelazJ) [reply source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: Does it create a new layer or change field type?

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst [hunk 3]
```diff
@@ -404,246 +146,150 @@ Parameters
...
-.......
+A new layer is created with the result of the expression.
 
-``output`` [vector]
-  <put output description here>
+The field calculator is very useful when used in :ref:`processing.modeler`.
 
-Console usage
-.............
+.. figure:: /static/user_manual/processing_algs/qgis/field_calculator.png
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: Is that really needed? It's a common dialog in QGIS (and if new things are added, it'd need to be updated here also.
If there was a screenshot to add, I'd vote for the refactor fields one (in practice).

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst [hunk 4]
```diff
@@ -404,246 +146,150 @@ Parameters
...
+  Add here the Python formula to evaluate. For example to calculate the area of
+  an input polygon layer you can add:
 
-``Output layer`` [vector]
-  <put output description here>
+  .. code-block:: python
 
-Console usage
-.............
+    value  = $geom.area()
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: is this a working code sample, out of the box? with `$geom`?

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst [hunk 5]
```diff
@@ -404,246 +146,150 @@ Parameters
...
 
 ``Field precision`` [number]
-  <put parameter description here>
+  Precision of the field. Useful with Float field type
 
-  Default: *3*
-
-``Create new field`` [boolean]
-  <put parameter description here>
+  Default: *0*
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: it was maybe my mistake. I also have 3

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst [hunk 6]
```diff
@@ -404,246 +146,150 @@ Parameters
...
 
-  Default: *True*
+``Global expression`` [string]
+  Optional.
 
-``Formula`` [string]
-  <put parameter description here>
+  Option to add a global expression
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: if found something in Anita's blog https://anitagraser.com/2013/04/20/advanced-python-field-calculator/. 
@anitagraser can I use your global expression description?

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst (hunk 4 above)
Comment (ghtmtt) [reply source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: it is and it works.. it is just a small example

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst [hunk 7]
```diff
@@ -404,246 +146,150 @@ Parameters
...
-``Output layer`` [vector]
-  <put output description here>
+  .. code-block:: python
 
-Console usage
-.............
+    value  = $geom.area()
 
-::
+  Default: *value =*
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: right.. I removed it.. it's quite better I think

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst [hunk 8]
```diff
@@ -404,246 +146,150 @@ Parameters
...
-------------------
 
-Description
-...........
+.. _qgis_dropfield:
 
-Lists unique values of an attribute table field and counts their number.
+Drop field(s)
+-------------
+Takes a vector layer and generates a new one that has the exact same content but
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: ahhah ;)

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst (hunk 3 above)
Comment (ghtmtt) [reply source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: right.. I replaced this picture with that of refactor field

File: source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst [hunk 9]
```diff
@@ -11,391 +11,133 @@ Vector table
...
 
-  processing.runalg('qgis:addfieldtoattributestable', input_layer, field_name, field_type, field_length, field_precision, output_layer)
+.. _qgis_add_field_to_attribute_table:
 
-See also
-........
-
-Advanced Python field calculator
---------------------------------
+Add field to attributes table
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply source/docs/user_manual/processing_algs/qgis/vector_table_tools.rst]: I agree on that. I think its better to have the same name even. Of course this has more sense to be changed in the app and then in the docs.

**Comment by ghtmtt** [comment]
@DelazJ Travis fails again for the ``code-block``. I'm going to merge it anyways as the local build gives no issues and I think that the failing is due to all warnings turned into errors.

**Pull request merged**